tlatools/org.lamport.tlatools/src/recomp/RecompVerify.java

```diff
@@ -47,151 +47,172 @@
+			timer.reset();
+			ltsProp = ParallelComposition.INSTANCE.parallel(ltsComp, ltsProp);
+			Utils.printVerbose(verbose, "New property gen (|| composition): " + timer.timeElapsed() + "ms");
+
+			// collect the alphabet
+			alphabetTester.update(tlcComp.actionsInSpec(), ltsProp);
+
+			// if the new safety property is TRUE or FALSE then model checking is done
+			if (SafetyUtils.INSTANCE.ltsIsSafe(ltsProp)) {
+				final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
+				printMsg = Utils.addPrint(printMsg,  "");
+				printMsg = Utils.addPrint(printMsg, "k: " + i);
+				printMsg = Utils.addPrint(printMsg,"Total # states checked: " + totalNumStatesChecked);
+				printMsg = Utils.addPrint(printMsg,"Property satisfied!");
+				if (!RVStrategy.isThreadDone()) System.out.print(printMsg);
+				return;
+			}
+			if (SafetyUtils.INSTANCE.hasErrInitState(ltsProp)) {
+				final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
+				Utils.printVerbose(verbose, "");
+				printMsg = Utils.addPrint(printMsg,"k: " + i);
+				printMsg = Utils.addPrint(printMsg,"Total # states checked: " + totalNumStatesChecked);
+				printMsg = Utils.addPrint(printMsg,"Property may be violated.");
+				//FSPWriter.INSTANCE.write(System.out, ltsProp);
+				if (!RVStrategy.isThreadDone()) System.out.print(printMsg);
+				return;
+			}
+		}
+
 		final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
-		Utils.printVerbose(verbose, "");
-		System.out.println("k: " + (components.size() - 1));
-		System.out.println("Total # states checked: " + totalNumStatesChecked);
-		System.out.println("Property may be violated.");
-		
-		// encode the sequence of actions that leads to an error in a new TLA+ file
-		// TODO write error trace for early termination
-		writeErrorTraceFile(tla, cfg, ltsProp);
-    	
-    	// not unix convention, but we use this to signal to the wrapper script that
-    	// it should produce an error trace
-    	System.exit(99);
-    }
-	
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "");
+		printMsg = Utils.addPrint(printMsg,"k: " + (components.size() - 1));
+		printMsg = Utils.addPrint(printMsg,"Total # states checked: " + totalNumStatesChecked);
+		printMsg = Utils.addPrint(printMsg,"Property may be violated.");
+
+		//Output everything
+		if (!RVStrategy.isThreadDone()) System.out.print(printMsg);
+
+		// encode the sequence of actions that leads to an error in a new TLA+ file	// encode the sequence of actions that leads to an error in a new TLA+ file
+		// TODO write error trace for early termination	// TODO write error trace for early termination
+		writeErrorTraceFile(tla, cfg, ltsProp);	writeErrorTraceFile(tla, cfg, ltsProp);
+
+		// not unix convention, but we use this to signal to the wrapper script that	// not unix convention, but we use this to signal to the wrapper script that
```

Comment:
please move this comment to line 62, since this comment is relevant to that line.